apps/emqx/src/emqx_sys_mon.erl

```diff
@@ -202,7 +202,8 @@ get_proc_lib_initial_call(Pid) ->
     end.
 
 portinfo(Port) ->
-    [{port, Port} | erlang:port_info(Port)].
+    PortInfo = [erlang:port_info(Port) || is_port(Port)],
+    [{port, Port} | PortInfo].
```

Comment:
this is a bit different than before.
before the change this function returns `[{port, Port} | PortInfo]`
now it returns `[{port, Port} | [PortInfo]]`

Reply:
`erlang:port_info/1` returns `undefined` when it is a port but closed.

Reply:
a safe impl is:
```
case is_port(Port) andalso erlang:port_info(Port) of
   L when is_list(L) -> L;
  _ -> []
end,
```

Reply:
In that case, the previous behavior was to create an improper list, right?

```erlang
6> [{port, hd(erlang:ports())} | undefined].
[{port,#Port<0.0>}|undefined]
```

Do we want to keep that?  Maybe we should check the result of `port_info/1` and return an empty list if closed?

Reply:
Ah, missed the last message due to lack of refresh.

I'll fix it to check the result then.